modules/gapi/src/api/s11n.cpp

```diff
@@ -44,6 +44,13 @@ std::vector<char> cv::gapi::serialize(const cv::GRunArgs& ra)
     return os.data();
 }
 
+std::vector<char> cv::gapi::serialize(const cv::GCompileArgs& ca)
+{
+    cv::gapi::s11n::ByteMemoryOutStream os;
+    serialize(os, ca);
```

Comment:
why not `os << ca` ?

Reply:
...there may be an overload.

Reply:
I actually didn't get why this additional level of indirection as serialize(os, ca) is required but it is how serialization for MetaArgs and RunArgs is done. I think I misunderstand something

Reply:
Let's see.

Reply:
Ok. The right answer is that while the serialization for vector may work, a generic deserialization for vector doesn't